docs/2.5. APIs - Query Parameters.md

```diff
@@ -12,60 +12,65 @@ Query APIs SHOULD support pagination of their API resources where the 'paged' tr
 
-* When query parameters which perform filtering are used at the same time as paging, the filters must be applied by the implementation before applying paging parameters to the resulting data set.
+When query parameters which perform filtering are used at the same time as paging, the filters MUST be applied by the implementation before applying paging parameters to the resulting data set.
 
-* A server may choose its own default value for paging limit (see Example 1).
+A server MAY choose its own default value for paging limit (see Example 1).
 
-* Where both 'since' and 'until' parameters are specified, the 'since' value takes precedence where a resulting data set is constrained by the server's value of 'limit' (see Example 5).
+Where both 'since' and 'until' parameters are specified, the 'since' value takes precedence where a resulting data set is constrained by the server's value of 'limit' (see Example 5).
 
-* Servers should include 'prev' and 'next' Link headers and may include 'first' and 'last' Link headers in all responses.
+Servers SHOULD include 'prev' and 'next' Link headers and MAY include 'first' and 'last' Link headers in all responses.
 
 ### Examples
 
-The following examples aim to identify how pagination should behave in the presence of a set of registered data. In order to avoid displaying full resource representations, the only data listed here is the 'update' timestamp associated with each registered record. The same procedures can be applied where 'creation' timestamps and the '?paging.order=create' parameter are used instead.
+The following examples show pagination for a set of registered data. In order to avoid displaying full resource representations, the only data listed here is the 'update' timestamp associated with each registered record. The same procedures can be applied where 'creation' timestamps and the '?paging.order=create' parameter are used instead.
 
 Where a paging limit is not specified in a request the server's default is used.
 
 **Sample Data: Registered Node Update Timestamps (Comma-Separated)**
-```
+
+```json
 [0:1, 0:2, 0:3, 0:4, 0:5, 0:6, 0:7, 0:8, 0:9, 0:10, 0:11, 0:12, 0:13, 0:14, 0:15, 0:16, 0:17, 0:18, 0:19, 0:20]
 ```
 
 Each of the above corresponds to the update timestamp of a corresponding Node, in the format &lt;seconds&gt;:&lt;nanoseconds&gt; and displayed in ascending order. These will be used throughout the following examples.
 
-Response payloads in the examples will show these values, but in a real implementation should be replaced by the corresponding JSON objects for the Nodes or other resources being queried.
+Response payloads in the examples will show these values, but in a real implementation SHOULD be replaced by the corresponding JSON objects for the Nodes or other resources being queried.
```

Comment:
If this were conformance language it would have to be MUST, which would look like a substantive change, so I suggest:

```suggestion
Response payloads in the examples will show these values, but in a real implementation would be replaced by the corresponding JSON objects for the Nodes or other resources being queried.
```

Reply:
Agreed. i thought that someone would say this when i wrote it.